solidity/test/bridge/VendingMachine.Upgrade.test.ts

```diff
@@ -0,0 +1,222 @@
+
+    // Deployment scripts deploy both `VendingMachine` and `TBTCVault` but they
+    // do not transfer the ownership of `TBTC` token to `TBTCVault`.
+    // We need to do it manually in tests covering `TBTCVault` behavior.
+    // Also, please note that `03_transfer_roles.ts` assigning `VendingMachine`
+    // upgrade initiator role to Keep Technical Wallet is skipped for Hardhat
+    // env deployment. That's why the upgrade initiator and `VendingMachine`
+    // owner is the deployer.
+    await vendingMachine
+      .connect(deployer)
+      .initiateVendingMachineUpgrade(tbtcVault.address)
+    await increaseTime(await vendingMachine.GOVERNANCE_DELAY())
+    await vendingMachine.connect(deployer).finalizeVendingMachineUpgrade()
+  })
+
+  describe("upgrade process", () => {
+    // Two accounts with 10 TBTC v1 each wrap their holdings to TBTC v2.
+    // See the main `before`.
+    const totalTbtcV1Balance = to1e18(20)
+
+    describe("step#1 - TBTC v1 transfer", () => {
+      it("should transfer all TBTC v1 to TBTCVault", async () => {
+        expect(await tbtcV1.balanceOf(vendingMachine.address)).to.equal(
+          to1e18(0)
+        )
+        expect(await tbtcV1.balanceOf(tbtcVault.address)).to.equal(
+          totalTbtcV1Balance
+        )
+      })
+    })
+
+    describe("step#2 - TBTC v1 withdrawal", () => {
+      it("should let the governance withdraw TBTC v1 from TBTCVault", async () => {
+        await tbtcVault
+          .connect(governance)
+          .recoverERC20(tbtcV1.address, governance.address, totalTbtcV1Balance)
+        expect(await tbtcV1.balanceOf(tbtcVault.address)).to.equal(0)
+        expect(await tbtcV1.balanceOf(governance.address)).to.equal(
+          totalTbtcV1Balance
+        )
+      })
+    })
+
+    describe("step#3 - BTC donation", () => {
+      before(async () => {
+        const data = SingleP2SHDeposit
+        const { fundingTx, reveal } = data.deposits[0] // it's a single deposit
+        reveal.vault = tbtcVault.address
+
+        // Simulate the wallet is a Live one and is known in the system.
+        await bridge.setWallet(reveal.walletPubKeyHash, {
+          ecdsaWalletID: ethers.constants.HashZero,
+          mainUtxoHash: ethers.constants.HashZero,
+          pendingRedemptionsValue: 0,
+          createdAt: await lastBlockTime(),
+          movingFundsRequestedAt: 0,
+          closingStartedAt: 0,
+          pendingMovedFundsSweepRequestsCount: 0,
+          state: walletState.Live,
+          movingFundsTargetWalletsCommitmentHash: ethers.constants.HashZero,
+        })
+
+        await bridge.revealDeposit(fundingTx, reveal)
+
+        relay.getCurrentEpochDifficulty.returns(data.chainDifficulty)
+        relay.getPrevEpochDifficulty.returns(data.chainDifficulty)
+
+        await bridge.submitDepositSweepProof(
+          data.sweepTx,
+          data.sweepProof,
+          data.mainUtxo,
+          tbtcVault.address
+        )
```

Comment:
NB: You can also make `runDepositSweepScenario` from `Bridge.Deposit.test.ts` exported and let it do the heavy lifting for you.